gst/nnstreamer/tensor_allocator.c

```diff
@@ -0,0 +1,119 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * Copyright (C) 2021 Junhwan Kim <jejudo.kim@samsung.com>
+ *
+ * @file    tensor_allocator.c
+ * @date    12 May 2021
+ * @brief   Allocator for memory alignment
+ * @author  Junhwan Kim <jejudo.kim@samsung.com>
+ * @see     http://github.com/nnstreamer/nnstreamer
+ * @bug     No known bugs
+ *
+ */
+
+#include <gst/gst.h>
+#include "nnstreamer_plugin_api.h"
+
+#define GST_TENSOR_ALLOCATOR "GstTensorAllocator"
+
+static gsize gst_tensor_allocator_alignment = 0;
+
+/**
+ * @brief struct for type GstTensorAllocator
+ */
+typedef struct
+{
+  GstAllocator parent;
+} GstTensorAllocator;
+
+/**
+ * @brief struct for class GstTensorAllocatorClass
+ */
+typedef struct
+{
+  GstAllocatorClass parent_class;
+} GstTensorAllocatorClass;
+
+
+static GType gst_tensor_allocator_get_type (void);
+G_DEFINE_TYPE (GstTensorAllocator, gst_tensor_allocator, GST_TYPE_ALLOCATOR);
+
+/**
+ * @brief   allocation wrapper that binds alignment parameter
+ */
+static GstMemory *
+_alloc (GstAllocator * allocator, gsize size, GstAllocationParams * params)
+{
+  GstAllocator *sysmem_alloc;
+  GstAllocatorClass *sysmem_aclass;
+  GstAllocationParams *_params;
+
+  sysmem_alloc = gst_allocator_find (GST_ALLOCATOR_SYSMEM);
+  sysmem_aclass = GST_ALLOCATOR_GET_CLASS (sysmem_alloc);
+  _params = gst_allocation_params_copy (params);
+  _params->align = gst_tensor_allocator_alignment;
+
+  return sysmem_aclass->alloc (allocator, size, _params);
+}
+
+/**
+ * @brief class initization for GstTensorAllocatorClass
+ */
+static void
+gst_tensor_allocator_class_init (GstTensorAllocatorClass * klass)
+{
+  GstAllocatorClass *allocator_class, *sysmem_aclass;
+  GstAllocator *sysmem_alloc;
+
+  allocator_class = (GstAllocatorClass *) klass;
+  sysmem_alloc = gst_allocator_find (GST_ALLOCATOR_SYSMEM);
+  sysmem_aclass = GST_ALLOCATOR_GET_CLASS (sysmem_alloc);
+
+  allocator_class->alloc = _alloc;
+  allocator_class->free = sysmem_aclass->free;
+}
+
+/**
+ * @brief initialzation for GstTensorAllocator
+ */
+static void
+gst_tensor_allocator_init (GstTensorAllocator * allocator)
+{
+  GstAllocator *sysmem_alloc, *alloc;
+
+  alloc = GST_ALLOCATOR_CAST (allocator);
+  sysmem_alloc = gst_allocator_find (GST_ALLOCATOR_SYSMEM);
+
+  alloc->mem_type = sysmem_alloc->mem_type;
```

Comment:
What happens if simply do ```alloc = system_alloc```?

Reply:
I guess `*alloc = *system_alloc` might work. Thx for the comment!

Reply:
Doesn't work by just copying it maybe because of some private values, FYI

Reply:
@kimjh12 In https://github.com/GStreamer/gstreamer/blob/master/gst/gstallocator.c, it seems that copying of allocator works. Anyway, at this point, let's merge it. But we need to more investigate it.

Reply:
Thanks for the detail. I' ll have a closer look